### ISSUE TITLE
Improve event notifications in general.

### ISSUE THREAD

**Issue by AkshayAgarwal007** [AkshayAgarwal007/Calendar#28]
Presently it simply uses BNotification and delivers an information notification to the user about the event's occurrence. But that may not be enough. The user may want to snooze the event for a certain period and want the notification again after that period. This is standard is almost all calendar apps and in my view a necessary feature.

**Comment by Perelandra0x309** [comment]
Unfortunately the built in BNotification in Haiku is very limited, as you note.  Notifications disappear after a certain amount of time elapses if the user does not click on it.  There is no notification history to view, so if you are away and a notification appears and disappears you will not know.  I did a little bit of work to improve that but you know, time.  Notifications may be improved sometime in the future, but for now if you are able to program something custom in the calendar app that would be best until BNotifications are improved.

Also there is no built-in scheduling service like cron yet.  So you may need to have Calendar open all the time, or maybe you will need to create a separate calendar daemon to display event reminders if Calendar is closed?

**Comment by CodeforEvolution** [comment]
If you need to launch an app on start up like @Perelandra0x309 said with a background daemon, you may want to take a look at Haiku's launch daemon: https://dev.haiku-os.org/wiki/LaunchDaemon
Hope this helps!

**Comment by AkshayAgarwal007** [comment]
@Perelandra0x309 Ya I spend quite some time looking at your work on implementing a notification center which looks cool. https://dev.haiku-os.org/ticket/12809

I was thinking of a separate background process which runs as a deskbar replicant and does the task of generating the reminders and an option to open Calendar similar to network status applet.

I'll have a look at it. @CodeforEvolution

**Comment by Perelandra0x309** [comment]
@AkshayAgarwal007 Yeah you could do the Deskbar replicant two ways.  Either the replicant itself contains the entire logic for keeping track of events, or you have a separate daemon which just uses a replicant as a basic user interface.  The replicant can pass messages back and forth with a daemon, I did that with my new notification center code.

**Comment by AkshayAgarwal007** [comment]
@Perelandra0x309 did you by chance work on the relative time formatting(of when the notification arrived...3 hours ago etc.) idea that humdingerb suggested? There doesn't seem to be a relative datetime formatter in locale kit that's why asked...

**Comment by Perelandra0x309** [comment]
@AkshayAgarwal007 I never did get to that point yet.

**Comment by AkshayAgarwal007** [comment]
I implemented a basic one after I asked you that... :)
 https://dev.haiku-os.org/ticket/13679
